### PR TITLE
Add a new package: graph500.

### PR DESCRIPTION
**Pull request by weijianwen**

### CODE REVIEW

File: var/spack/repos/builtin/packages/graph500/package.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+# conditions of the GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+from spack import *
+
+
+class Graph500(Package):
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/graph500/package.py]: This should be rewritten to use `MakefilePackage`. You can set `build_directory = 'src'` and override the `edit` and `install` phases.

File: var/spack/repos/builtin/packages/graph500/package.py [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    homepage = "https://graph500.org"
+    url      = "https://github.com/graph500/graph500/archive/graph500-3.0.0.tar.gz"
+
+    version('3.0.0', 'a2ebb4783b21e2d86387a217776395e3')
+
+    depends_on('mpi@2.0:')
+
+    build_directory = 'src'
+
+    def eidt(self):
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/graph500/package.py]: Typo: eidt

**Comment by weijianwen** [comment]
Convert it to a MakefilePackage.

File: var/spack/repos/builtin/packages/graph500/package.py [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+        edit = FileFilter('src/Makefile')
+        edit.filter(r'^MPICC\s*=.*', 'MPICC={0}'.format(spec['mpi'].mpicc))
+
+    def install(self, spec, prefix):
+        with working_dir('src'):
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/graph500/package.py]: You can replace `'src'` with `self.build_directory` so you don't need to use as many ["magic numbers"](https://en.wikipedia.org/wiki/Magic_number_(programming))

File: var/spack/repos/builtin/packages/graph500/package.py (hunk 2 above)
Comment (weijianwen) [reply var/spack/repos/builtin/packages/graph500/package.py]: Fixed.

File: var/spack/repos/builtin/packages/graph500/package.py (hunk 3 above)
Comment (weijianwen) [reply var/spack/repos/builtin/packages/graph500/package.py]: Fixed.

**Pull request merged**